### ISSUE TITLE
Some Error with v2 api format

### ISSUE THREAD

**Issue by antoncoding** [NoteGio/openrelay#121]
I was trying to use the /v2 endpoint with the new v2 0x/connect library (on Kovan Testnet) and I encounter some Errors:
The `response` i got from [getOrderConfigAsync](https://0xproject.com/docs/connect/2.0.0#HttpClient-getOrderConfigAsync)() function was incorrect:
Response from OpenRelay:
```
feeRecipient: "0xc22d5b2951db72b44cfb8089bb8cd374a3c354ea",
makerFee: BigNumber {s: 1, e: 17, c: Array(1)},
sender: "0x0000000000000000000000000000000000000000",
takerFee: BigNumber {s: 1, e: 0, c: Array(1)},
takerToSpecify: "0x0000000000000000000000000000000000000000"
```
while the `sender` should be `senderAddress` and `feeRecipient` should be `feeRecipientAddress`, otherwise the response will not conform with the standard scheme.

Another error is when I submit my order with httpclient.[submitOrderAsync](https://0xproject.com/docs/connect/2.0.0#HttpClient-submitOrderAsync)(signedOrder), I got the following error: 
```
{
"code":100,"reason":"Validation Failed",
"validationErrors":[
   {
   "field":"exchangeContractAddress",
   "code":1002,
   "reason":"Unknown exchangeContractAddress"
   }
]
}
```
I got my `contractAddress` value in `signedOrder` by calling `Web3WrapperInstance.getAvailableAddressesAsync()`, which is `0x35dd2932454449b14cee11a94d3674a936d5d7b2`
I'm not sure why the address is incorrect , I also tried change my `contractAddress` key in my `signedOrder` to `exchangeContractAddress` but still got the same error.

**Comment by AusIV** [comment]
Thanks for pointing these out. They'll be addressed by https://github.com/NoteGio/openrelay/issues/121

As far as the Kovan address - we supported an older Kovan iteration of the v2 contract; the one you use was updated after they deployed to mainnet. I thought that was already updated, but if not it will go in with the order config updates.

**Comment by ochimo** [comment]
Related question - when using Ropsten and retrieving contract address `zeroEx.exchange.getContractAddress()` I am getting `0x4530c0483a1633c7a1c97d2c53721caff2caaaaf` which is correct as defined in [https://0xproject.com/wiki#Deployed-Addresses](url) for v2.

When doing post on https://api.openrelay.xyz/v2/order I am also getting back error `Unknown exchangeContractAddress`

In branch 0xv2-support -> openrelay/ingest/ingest.go the list of knownExchanges line 59 seems to be incorrect - the first two exchanges are accurate for mainnet & kovan -- but Ropsten address isn't listed.  Is this a bug or something I'm missing?

**Comment by AusIV** [comment]
openrelay.xyz doesn't currently support v2 on Ropsten. We don't have the monitoring services etc. in place to prune ropsten orders from the v2 orderbook.

That list of knownExchanges on line 59 of ingest.go ought to be replaced with a database lookup, now that we track supported exchanges in the database, but I haven't had time to prioritize that effort. That would at least people running the open source version of openrelay to support exchanges we don't have the services for on openrelay.xyz. I'll put together a ticket for that.

**Comment by AusIV** [comment]
@antoncoding's issue should be addressed in the latest deployment. @ochimo's point is being addressed by #123 to remove the hardcoded dependency in the open source project, and #124 to support ropsten on openrelay.xyz

**Comment by ochimo** [comment]
@AusIV quick question, if I were to bring up my own instance of OpenRelay and modified it to accept the Ropsten exchange contract would it work?  I am just trying to test full lifecycle with it and am happy to manage my own order book while I have the server up.  

Since I'm new to this I'm wondering if I'm missing something before jumping in this direction.  Thanks -
